modules/aggregation-layers/src/new-grid-layer/new-grid-layer.js

```diff
@@ -0,0 +1,155 @@
+import {CompositeLayer} from '@deck.gl/core';
+import {AGGREGATION_OPERATION} from '../utils/gpu-grid-aggregation/gpu-grid-aggregator-constants';
+import GPUGridAggregator from '../utils/gpu-grid-aggregation/gpu-grid-aggregator';
+import GPUGridLayer from '../gpu-grid-layer/gpu-grid-layer';
+import GridLayer from '../grid-layer/grid-layer';
+
+const defaultProps = Object.assign({}, GPUGridLayer.defaultProps, GridLayer.defaultProps);
+
+// Function to convert from getWeight/aggregation props to getValue prop for Color and Elevation
+function getMean(pts, accessor) {
+  const filtered = pts.map(item => accessor(item)).filter(pt => Number.isFinite(pt));
+
+  return filtered.length ? filtered.reduce((accu, curr) => accu + curr, 0) / filtered.length : null;
+}
+
+function getSum(pts, accessor) {
+  const filtered = pts.map(item => accessor(item)).filter(pt => Number.isFinite(pt));
+
+  return filtered.length ? filtered.reduce((accu, curr) => accu + curr, 0) : null;
+}
+
+function getMax(pts, accessor) {
+  const filtered = pts.map(item => accessor(item)).filter(pt => Number.isFinite(pt));
+
+  return filtered.length
+    ? filtered.reduce((accu, curr) => (curr > accu ? curr : accu), -Infinity)
+    : null;
+}
+
+function getMin(pts, accessor) {
+  const filtered = pts.map(item => accessor(item)).filter(pt => Number.isFinite(pt));
+
+  return filtered.length
+    ? filtered.reduce((accu, curr) => (curr < accu ? curr : accu), Infinity)
+    : null;
+}
+
+function getValueFunc(aggregation, accessor) {
+  switch (aggregation) {
+    case AGGREGATION_OPERATION.MIN:
+      return pts => getMin(pts, accessor);
+    case AGGREGATION_OPERATION.SUM:
+      return pts => getSum(pts, accessor);
+    case AGGREGATION_OPERATION.MEAN:
+      return pts => getMean(pts, accessor);
+    case AGGREGATION_OPERATION.MAX:
+      return pts => getMax(pts, accessor);
+    default:
+      return null;
+  }
+}
+
+export default class NewGPULayer extends CompositeLayer {
+  initializeState() {
+    this.state = {
+      useGPUAggregation: true
+    };
+  }
+
+  getPickingInfo(opts) {
+    // TODO: implement picking.
+    return Object.assign({}, opts.info, {picked: false, object: null});
+  }
+
+  updateState({oldProps, props, changeFlags}) {
+    const newState = {};
+    newState.useGPUAggregation = this.canUseGPUAggregation(props);
+    this.setState(newState);
+  }
+
+  renderLayers() {
+    let gridLayer = null;
+
+    if (this.state.useGPUAggregation) {
+      gridLayer = new GPUGridLayer(
+        this.props,
+        this.getSubLayerProps({
```

Comment:
Lacking support for update triggers?

Reply:
Nothing happening in this layer, no attribute calculations or aggregation. It basically checking if aggregation can happen on GPU or not. `GridLayer` and `GPUGridLayer` have update triggers setup.

Reply:
You still need to pass the update triggers to the sublayers. Here the update triggers in `this.props` is overwritten by `getSubLayerProps`.

Reply:
Done, also update the document page for `getSublayerProps` , removed `Optional` for `updateTriggers` argument. Seems like we always need to pass this to avoid overriding.